Framework/Kernel/src/TimeSeriesProperty.cpp

```diff
@@ -541,209 +541,136 @@ void TimeSeriesProperty<TYPE>::splitByTime(
-    while (continue_add) {
-      size_t inner_while_counter = 0;
-      if (index_tsp_time == tspTimeVecSize) {
-        // last entry. quit all loops
-        continue_add = false;
-        continue_search = false;
-        partial_target_filled = true;
-        break;
-      }
+    if (timeIndex > 0)
+      --timeIndex;
+
+    // add the continuous entries to same wsIndex time series property
+    const size_t numEntries = currentTimes.size();
 
-      // add current entry
-      if (outputs[target]->size() == 0 ||
-          outputs[target]->lastTime() < tsp_time_vec[index_tsp_time]) {
+    // Put TSP's entries into the corresponding ws index
+    for (; timeIndex < numEntries; timeIndex++) {
+      if (output[wsIndex]->size() == 0 ||
+          output[wsIndex]->lastTime() < currentTimes[timeIndex]) {
         // avoid to add duplicate entry
-        outputs[target]->addValue(m_values[index_tsp_time].time(),
-                                  m_values[index_tsp_time].value());
+        output[wsIndex]->addValue(m_values[timeIndex].time(),
+                                  m_values[timeIndex].value());
       }
 
-      const size_t nextTspIndex = index_tsp_time + 1;
-      if (nextTspIndex < tspTimeVecSize) {
-        if (tsp_time_vec[nextTspIndex] > split_stop_time) {
-          // next entry is out of this splitter: add the next one and quit
-          if (outputs[target]->lastTime() < m_values[nextTspIndex].time()) {
-            // avoid the duplicate cases occurred in fast frequency issue
-            outputs[target]->addValue(m_values[nextTspIndex].time(),
-                                      m_values[nextTspIndex].value());
-          }
-          // FIXME - in future, need to find out WHETHER there is way to
-          // skip the
-          // rest without going through the whole sequence
-          continue_add = false;
+      const size_t nextEntryIndex = timeIndex + 1;
+      if (nextEntryIndex < numEntries &&
+          currentTimes[nextEntryIndex] > filterEndTime) {
+        // We are beyond the final filter time
+        if (output[wsIndex]->lastTime() < m_values[nextEntryIndex].time()) {
+          // avoid the duplicate cases occurred in fast frequency issue
+          output[wsIndex]->addValue(m_values[nextEntryIndex].time(),
+                                    m_values[nextEntryIndex].value());
         }
+        ++timeIndex;
+        break;
       }
+    } // End of adding properties to current ws index
 
-      // advance to next entry
-      ++index_tsp_time;
-
-      ++inner_while_counter;
-    } // END-WHILE continue add
-
-    // make splitters to advance to next
-    ++index_splitter;
-    if (index_splitter == splitter_time_vec.size() - 1) {
-      // already last splitters
-      continue_search = false;
-    } else {
-      split_start_time = split_stop_time;
-      split_stop_time = splitter_time_vec[index_splitter + 1];
+    if (timeIndex == numEntries) {
+      destinationPartiallyFilled = true;
```

Comment:
I think the loop previously used to break when it hit this condition?

I have a suggestion for further refactoring here to consolidate the next loop into this one, making this flag obsolete. I'll push the changes to this branch.

Reply:
This is much clearer; I was struggling to understand and refactor this block so the commit helps significantly.